core/src/polling_monitor/ipfs_service.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow::{anyhow, Error};
+use bytes::Bytes;
+use cid::Cid;
+use futures::{Future, FutureExt};
+use graph::{cheap_clone::CheapClone, ipfs_client::IpfsClient, tokio::sync::Semaphore};
+use std::{pin::Pin, sync::Arc, task::Poll, time::Duration};
+use tower::Service;
+
+const CLOUDFLARE_TIMEOUT: u16 = 524;
+const GATEWAY_TIMEOUT: u16 = 504;
+
+#[derive(Clone)]
+pub struct IpfsService {
+    client: IpfsClient,
+    max_file_size: u64,
```

Comment:
Do we know how much this will be?

Reply:
For our other IPFS uses we use an env var `GRAPH_MAX_IPFS_FILE_BYTES` that defaults to 256MB. I'll probably just start with that when IpfsService needs to be initialized somewhere.